File: 백준/Silver/1978. 소수 찾기/소수 찾기.py
N= int(input())
flo = map(int,input().split())
answer=0
for i in flo:
    error = 0
    if i>1:
        for k in range(2,i):
            if i%k==0:
                error +=1
        if error ==0:
            answer+=1

print(answer)

File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] array = new int[n+1];
        for (int i = 2; i <=n ; i++) {
            array[i]=i;
        }
        for (int i = 2; i*i <=n ; i++) {
            if (array[i]==0)continue;
            for (int j = i*i; j <=n ; j+=i) {
                array[j]=0;
            }
        }
        for (int i = 0; i <= n ; i++) {
            if (array[i]>0){
                answer+=1;
            }
        }
        return answer;
    }
}